From 7e79455e907d1184d8708278b39bfa30251ceddf Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 9 Apr 2012 14:20:50 +0900
Subject: [PATCH 20/97] ext/dfbvideosink: compensate for the interpretation
 difference of colorspace definitions

The interpretation of colorspace definitions are different between
dfbvideosink and libshvio.
XRGB32 is indicated by following definitions each other.
- dfbvideosink: DSPF_RGB32
- libshvio: REN_XRGB32

This patch compensates for this difference when the definitions are
converted in dfb2shvio_format().
---
 ext/directfb/dfbvideosink.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index 2966626..1208f90 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1586,7 +1586,9 @@ dfb2shvio_format (DFBSurfacePixelFormat dfbfmt)
     case DSPF_RGB24:
       return REN_RGB24;
     case DSPF_RGB32:
-      return REN_RGB32;
+      /* compensate for an interpretation difference of colorspace definitions
+         between directfb and libshvio */
+      return REN_XRGB32;
     case DSPF_ARGB:
       return REN_ARGB32;
     default:
-- 
1.7.10.4

